pytensor/tensor/slinalg.py

```diff
@@ -1669,6 +1670,73 @@ def block_diag(*matrices: TensorVariable):
     return _block_diagonal_matrix(*matrices)
 
 
+class BandedDot(Op):
+    __props__ = ("lower_diags", "upper_diags")
+    gufunc_signature = "(m,n),(n)->(m)"
+
+    def __init__(self, lower_diags, upper_diags):
+        self.lower_diags = lower_diags
+        self.upper_diags = upper_diags
+
+    def make_node(self, A, x):
+        A = as_tensor_variable(A)
+        x = as_tensor_variable(x)
+
+        out_dtype = pytensor.scalar.upcast(A.dtype, x.dtype)
```

Comment:
wrong for integers/should raise. Also reject  complex?

Reply:
I copied this from other make_node in slinalg (eigvalsh, eigvalsh grad, solve lyapunov stuff). What's the right way to upcast here?

Reply:
The right way is to predict what scipy outputs. Some Ops are lazy and just call scipy with a minimal input case to find out the output type. I don't love that.

Which makes me wonder I guess numba/direct call to xbmv doesn't work with integers arrays, so we may need to cast/raise?